pkg/cli/cluster_test.go

```diff
@@ -0,0 +1,152 @@
+package cli
+
+import (
+	"crypto/x509"
+	"testing"
+	"time"
+
+	"github.com/dcos/dcos-cli/pkg/config"
+	"github.com/spf13/afero"
+	"github.com/stretchr/testify/require"
+)
+
+func TestGetters(t *testing.T) {
+	conf := config.Empty()
+
+	conf.Set("core.dcos_url", "https://dcos.example.com")
+	conf.Set("core.dcos_acs_token", "token_zj8Tb0vhQw")
+	conf.Set("core.ssl_verify", "/path/to/dcos_ca.crt")
+	conf.Set("core.timeout", 15)
+	conf.Set("cluster.name", "mr-cluster")
+
+	cluster := NewCluster(conf)
+
+	require.Equal(t, "https://dcos.example.com", cluster.URL())
+	require.Equal(t, "token_zj8Tb0vhQw", cluster.ACSToken())
+	require.Equal(t, true, cluster.TLS().Insecure)
+	require.Equal(t, 15*time.Second, cluster.Timeout())
+	require.Equal(t, "mr-cluster", cluster.Name())
+}
+
+func TestSetters(t *testing.T) {
+	conf := config.Empty()
+
+	cluster := NewCluster(conf)
+	cluster.SetURL("https://dcos.example.com")
+	cluster.SetACSToken("token_XYZ")
+	cluster.SetTLS(TLS{})
+	cluster.SetTimeout(15 * time.Second)
+	cluster.SetName("custom-cluster-name")
+
+	require.Equal(t, "https://dcos.example.com", conf.Get("core.dcos_url"))
+	require.Equal(t, "token_XYZ", conf.Get("core.dcos_acs_token"))
+	require.Equal(t, "true", conf.Get("core.ssl_verify"))
+	require.EqualValues(t, 15, conf.Get("core.timeout"))
+	require.Equal(t, "custom-cluster-name", conf.Get("cluster.name"))
+}
+
+func TestTLSToString(t *testing.T) {
+	expectedTLSStrings := []struct {
+		tls TLS
+		str string
+	}{
+		{TLS{}, "true"},
+		{TLS{Insecure: true}, "false"},
+		{TLS{RootCAsPath: "/path/to/ca"}, "/path/to/ca"},
+	}
+
+	for _, exp := range expectedTLSStrings {
+		require.Equal(t, exp.str, exp.tls.String())
+	}
+}
+
+func TestGetTLS(t *testing.T) {
+
+	expectedTLSInsecureSkipVerify := []struct {
+		val      interface{}
+		insecure bool
+	}{
+		{"True", false},
```

Comment:
I've tested values which make sense to me, `T` or `F` for eg. is very uncommon. For simplicity we've relied on ParseBool for now but I think we should move away from this and define what we want instead of depending on what a language does.

This is a topic I'd like to bring to the UX team, see https://jira.mesosphere.com/browse/DCOS-11460 for example.